app/models/order.rb

```diff
@@ -13,6 +14,18 @@ class Order < ActiveRecord::Base
   validates :price, :quantity, numericality: { greater_than: 0 }
   validates :type, inclusion: { in: %w(BuyOrder SellOrder) }
 
+  def cancel!
```

Comment:
https://www.ruby-forum.com/topic/176830#773946

> The bang (!) does not mean "destructive" nor lack of it mean non
destructive either.  The bang sign means "the bang version is more
dangerous than its non bang counterpart; handle with care".  Since
Ruby has a lot of "destructive" methods, if bang signs follow your
opinion, every Ruby program would be full of bangs, thus ugly.

> matz.

If we take matz's word as law, this method should probably be called `cancel` unless there is a less dangerous alternative.

What do you think?